gsa/src/gmp/__tests__/model.js

```diff
@@ -0,0 +1,143 @@
+/* Greenbone Security Assistant
+ *
+ * Authors:
+ * Steffen Waterkamp <steffen.waterkamp@greenbone.net>
+ *
+ * Copyright:
+ * Copyright (C) 2018 Greenbone Networks GmbH
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+import Model from 'gmp/model';
+import {isDate} from 'gmp/models/date';
+import Tag from 'gmp/models/tag';
+
+describe('Model tests', () => {
+
+  test('end_time is parsed correctly', () => {
+    const elem = {
+      end_time: '2018-10-10T11:41:23.022Z',
+    };
+    const model = new Model(elem);
+    expect(model.endTime).toBeDefined();
+    expect(model.end_time).toBeUndefined();
+    expect(isDate(model.endTime)).toBe(true);
+  });
+
+  test('permissions are parsed correctly', () => {
+    const elem = {
+      permissions: {
+        permission: [
+          {name: 'everything'},
+          {name: 'may_foo'},
+        ],
+      },
+    };
+    const model = new Model(elem);
+    expect(model.userCapabilities).toBeDefined();
+    expect(model.user_capabilities).toBeUndefined();
+    expect(model.userCapabilities._capabilities)
+      .toEqual(new Set(['everything', 'may_foo']));
```

Comment:
This tests internal behavior of Capabilities. Therefore it should be replaced with something like
```js
 expect(model.userCapabilities.length).toEqual(2);
```

If you use real permissions like `'get_foo'` you could also test for
```js
 expect(model.userCapabilities.mayAccess('foo')).toEqual(true);
```